warehouse/legacy/api/xmlrpc/__init__.py

```diff
@@ -10,4 +10,21 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from pyramid.events import NewResponse
+
 import warehouse.legacy.api.xmlrpc.views  # noqa
+
+
+def on_new_response(new_response_event):
+    request = new_response_event.request
+    if not hasattr(request, 'content_length_metric_name'):
+        return
+
+    request.registry.datadog.histogram(
+        request.content_length_metric_name,
+        new_response_event.response.content_length
+    )
+
+
+def includeme(config):
+    config.add_subscriber(on_new_response, NewResponse)
```

Comment:
will this affect all responses? is there any way to only apply this to the xmlrpc views?

Reply:
aye, but we're still going to be executing the logic on every request.

this is cleaner for a metric we might be submitting with every request, but I'm wondering what it really gains us over submitting the metric inline.

Reply:
The subscriber is going to bail pretty quickly if that attribute doesn't exist on the request, I don't think it's much overhead. We're doing a similar thing already here: https://github.com/pypa/warehouse/blob/0e33d86537ce10469ec93e8fd15532b16f579dea/warehouse/datadog/pyramid_datadog.py#L105-L108

The reason I did it via a subscriber is because it's difficult to get the content length of the response in the view, because it just returns a `list` of results, not an actual `Response`.

The alternative would be either a) estimating the actual `Content-Length` in the view (messy/unnecessarily expensive to serialize the payload twice) , or b) use character length as a proxy for content length (which is not quite right).

Reply:
Hmm, perhaps this could be done with a response callback instead... (https://docs.pylonsproject.org/projects/pyramid/en/latest/narr/hooks.html#using-response-callbacks)